packages/lib/server/username.ts

```diff
@@ -116,14 +116,82 @@ const usernameCheck = async (usernameRaw: string) => {
   const username = slugify(usernameRaw);
 
   const user = await prisma.user.findFirst({
-    where: { username, organizationId: null },
+    where: {
+      username,
+      // Simply remove it when we drop organizationId column
+      organizationId: null,
+    },
+    select: {
+      id: true,
+      username: true,
+    },
+  });
+
+  if (user) {
+    response.available = false;
+  }
+
+  if (await isPremiumUserName(username)) {
+    response.premium = true;
+  }
+
+  // get list of similar usernames in the db
+  const users = await prisma.user.findMany({
+    where: {
+      username: {
+        contains: username,
+      },
+    },
+    select: {
+      username: true,
+    },
+  });
+
+  // We only need suggestedUsername if the username is not available
+  if (!response.available) {
+    response.suggestedUsername = await generateUsernameSuggestion(
+      users.map((user) => user.username).filter(notEmpty),
+      username
+    );
+  }
+
```

Comment:
Reverted usernameCheck back to it's previous state and introduced `usernameCheckForSignup` for the signup specific use case where an already created user is being allowed to signup so that he can set his password. Note that the user is created as soon as he is invited.